test/common/protobuf/utility_test.cc

```diff
@@ -261,6 +263,364 @@ TEST_F(ProtobufUtilityTest, LoadTextProtoFromFile_Failure) {
                                 "\" as a text protobuf (type envoy.config.bootstrap.v2.Bootstrap)");
 }
 
+TEST_F(ProtobufUtilityTest, Redact) {
+  // Message to redact: contains a bunch of sensitive data.
+  envoy::test::Sensitive actual;
+  TestUtility::loadFromYaml(R"EOF(
```

Comment:
I'd suggest breaking this into a bunch of smaller tests for the different patterns. I think it'll be easier to follow and analyze failures in the future should this break at some point.

Reply:
Yeah, that's fair, I was being lazy.